gpkit/tests/t_sub.py

```diff
@@ -43,6 +43,17 @@ def test_string_mutation(self):
                                    1.0)
         self.assertEqual(x.sub("x", x), x)
 
+    def test_quantity_sub(self):
+        if gpkit.units:
+            x = Variable("x", 1, "cm")
+            y = Variable("x", 1)
+            self.assertEqual(x.sub(x, 1*gpkit.units.m).c.magnitude, 100)
+            self.assertRaises(ValueError, x.sub, x, 1*gpkit.units.N)
+            self.assertRaises(ValueError, y.sub, y, 1*gpkit.units.N)
+            v = gpkit.VectorVariable(3, "v", "cm")
+            subbed = v.sub(v, [1, 2, 3]*gpkit.units.m)
+            self.assertEqual([z.c.magnitude for z in subbed], [100, 200, 300])
+
```

Comment:
I mean, it doesn't have to be this way, but right now we're still permitting float subs.